### PR TITLE
Add SimpleRewarder contract

### PR DESCRIPTION
**Pull request by penandlim**
Allows one additional reward

### CODE REVIEW

File: contracts/rewards/SimpleRewarder.sol [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+     */
+    modifier onlyMiniChef() {
+        require(msg.sender == MINICHEF, "Rewarder: caller is not MiniChef");
+        _;
+    }
+
+    /**
+     * @notice Serves as the constructor for clones, as clones can't have a regular constructor.
+     * Initializes state varialbes with the given parameter.
+     * @param data abi encoded data in format of (IERC20 rewardToken, address owner, uint256 rewardPerSecond, IERC20 masterLpToken, uint256 pid).
```
Comment (pdyraga) [inline contracts/rewards/SimpleRewarder.sol:99]: Would love to confirm the parameters:
- `rewardToken` is KEEP token (`0x85Eee30c52B0b379b046Fb0F85F4f3Dc3009aFEC`)
- `owner` can be DAO or community multisig, as the only responsibility of the owner is to `setRewardPerSecond`
- `rewardPerSecond` should be `413359788359788360` assuming 250k KEEP is allocated weekly
- `masterLpToken` is `0xB4C35747c26E4aB5F1a7CdC7E875B5946eFa6fa9` which is [`
LPRewardsTBTCv2Saddle` Synthetix Unipool](https://etherscan.io/address/0xb4c35747c26e4ab5f1a7cdc7e875b5946efa6fa9#code) contract for TBTC V2 + SADDLE Meta V2
- `pid` is something MiniChef contract provides (?)

File: contracts/rewards/SimpleRewarder.sol [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        uint64 lastRewardTime,
+        uint256 lpSupply,
+        uint256 accToken1PerShare
+    );
+    event LogRewardPerSecond(uint256 rewardPerSecond);
+    event LogInit(
+        IERC20 indexed rewardToken,
+        address owner,
+        uint256 rewardPerSecond,
+        IERC20 indexed masterLpToken
```
Comment (pdyraga) [inline contracts/rewards/SimpleRewarder.sol:78]: Nitpicking: `rewardToken` and `masterLpToken` look like something that never changes so they do not necessarily need to be indexed.

File: contracts/rewards/SimpleRewarder.sol [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+            data,
+            (IERC20, address, uint256, IERC20, uint256)
+        );
+        require(rewardToken != IERC20(0), "Rewarder: bad rewardToken");
+        require(
+            IMiniChef(MINICHEF).lpToken(pid) == masterLpToken,
+            "Rewarder: bad pid or masterLpToken"
+        );
+        emit LogInit(rewardToken, owner, rewardPerSecond, masterLpToken);
+    }
```
[the remark this answers is not in the document]
Comment (penandlim) [reply contracts/rewards/SimpleRewarder.sol:115]: Changed from directly setting the `owner` storage variable to calling `transferOwnership()`.

File: contracts/rewards/SimpleRewarder.sol [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+     * @notice Modifier to restrict caller to be only MiniChef
+     */
+    modifier onlyMiniChef() {
+        require(msg.sender == MINICHEF, "Rewarder: caller is not MiniChef");
+        _;
+    }
+
+    /**
+     * @notice Serves as the constructor for clones, as clones can't have a regular constructor.
+     * Initializes state varialbes with the given parameter.
```
Comment (alphastorm) [inline contracts/rewards/SimpleRewarder.sol:98]: nit: s/varialbes/variables

File: contracts/rewards/SimpleRewarder.sol [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+            accToken1PerShare = accToken1PerShare.add(
+                reward.mul(ACC_TOKEN_PRECISION) / lpSupply
+            );
+        }
+        pending = (user.amount.mul(accToken1PerShare) / ACC_TOKEN_PRECISION)
+            .sub(user.rewardDebt);
+    }
+
+    /**
+     * @notice Returns pending reward tokens addresses and reward amounts for given address,
```
Comment (alphastorm) [inline contracts/rewards/SimpleRewarder.sol:179]: nit: s/address,/address.

**Review by alphastorm — APPROVED** [review]

**Pull request merged**